public/app/features/trails/MetricSelect/MetricSelectScene.tsx

```diff
@@ -220,45 +222,49 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> i
       return;
     }
 
-    const matchTerms: string[] = [];
+    const filters: AdHocVariableFilter[] = [];
 
     const filtersVar = sceneGraph.lookupVariable(VAR_FILTERS, this);
-    const hasFilters = filtersVar instanceof AdHocFiltersVariable && filtersVar.getValue()?.valueOf();
-    if (hasFilters) {
-      matchTerms.push(sceneGraph.interpolate(trail, '${filters}'));
+    const adhocFilters = filtersVar instanceof AdHocFiltersVariable ? (filtersVar?.state.filters ?? []) : [];
+    if (adhocFilters.length > 0) {
+      filters.push(...adhocFilters);
     }
 
     const metricSearchRegex = createPromRegExp(trail.state.metricSearch);
     if (metricSearchRegex) {
-      matchTerms.push(`__name__=~"${metricSearchRegex}"`);
+      filters.push({
+        key: '__name__',
+        operator: '=~',
+        value: metricSearchRegex,
+      });
     }
 
     let noOtelMetrics = false;
     let missingOtelTargets = false;
 
-    if (trail.state.useOtelExperience) {
-      const jobsList = trail.state.otelTargets?.jobs;
-      const instancesList = trail.state.otelTargets?.instances;
-      // no targets have this combination of filters so there are no metrics that can be joined
-      // show no metrics
-      if (jobsList && jobsList.length > 0 && instancesList && instancesList.length > 0) {
-        const otelMatches = limitOtelMatchTerms(matchTerms, jobsList, instancesList, missingOtelTargets);
-
-        missingOtelTargets = otelMatches.missingOtelTargets;
-
-        matchTerms.push(otelMatches.jobsRegex);
-        matchTerms.push(otelMatches.instancesRegex);
-      } else {
-        noOtelMetrics = true;
-      }
-    }
+    // TODO: fix this
+    // if (trail.state.useOtelExperience) {
+    //   const jobsList = trail.state.otelTargets?.jobs;
+    //   const instancesList = trail.state.otelTargets?.instances;
+    //   // no targets have this combination of filters so there are no metrics that can be joined
+    //   // show no metrics
+    //   if (jobsList && jobsList.length > 0 && instancesList && instancesList.length > 0) {
+    //     const otelMatches = limitOtelMatchTerms(matchTerms, jobsList, instancesList, missingOtelTargets);
+    //
+    //     missingOtelTargets = otelMatches.missingOtelTargets;
+    //
+    //     matchTerms.push(otelMatches.jobsRegex);
+    //     matchTerms.push(otelMatches.instancesRegex);
+    //   } else {
+    //     noOtelMetrics = true;
+    //   }
+    // }
```

Comment:
@bohandley You might want to check this.

Reply:
Pushed update to fix  this.